Blizzard.IO.RabbitMQ/Rpc/NetqRabbitRpcConnection.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Blizzard.IO.RabbitMQ.Entities.Rpc;
+using EasyNetQ;
+using EasyNetQ.Producer;
+using System;
+using ISerializer = Blizzard.IO.Core.Rpc.ISerializer;
+
+namespace Blizzard.IO.RabbitMQ.Rpc
+{
+    public class NetqRabbitRpcConnection : INetqRpcRabbitConnection<Func<Type, object>>
+    {
+        public IBus RabbitBus { get; }
+        public RpcMessageType MessageType { get; }
+
+        public NetqRabbitRpcConnection(RpcConfiguration configuration, string hostname, string username, string password, int heartBeat = 10, int preFetch = 50, ushort timeout = 10,
+            bool publisherConfirms = false, bool persistent = true, string product = null, string platform = null, string virtualHost = null,
+            ISerializer serializer = null, RpcMessageType rpcMessageType = RpcMessageType.Concrete)
+        {
+            MessageType = rpcMessageType;
+            string connectionString = $"host={hostname};username={username};password={password};requestedHeartbeat={heartBeat};prefetchcount={heartBeat};" +
+                $"persistentMessages={persistent};publisherConfirms={publisherConfirms}";
+            if (product != null)
+            {
+                connectionString += $"product={product}";
+            }
+            if (platform != null)
+            {
+                connectionString += $"platform={platform}";
+            }
+            if (virtualHost != null)
+            {
+                connectionString += $"virtualHost={virtualHost}";
+            }
```

Comment:
There is a way to integrate all of these optional properties into the `connectionString`, example on `product`:
`{(product != null ? $"product={product}" : string.Empty)}`

Reply:
It is not good to include the property with empty value in the connection string

Reply:
If product is null, nothing will be interpolated into the string because of `string.Empty`.
If product have value, the whole inner interpolation `$"product={product}"` will be integrated into the string, there's no way youll get something like `...;product=;...`  inside your `connectionString`

Reply:
Any way I think this much less readable than a few simple if's

Reply:
Maybe, but i this way we look more like we're hardcore c# boys
I dont care, decide youself and note an `f` for me to resolve this

Reply:
I think it looks more readable with the ifs
so f for ifs

Reply:
I went with my so the non hardcore c# boys could read it too